homeassistant/helpers/entity.py

```diff
@@ -253,8 +254,35 @@ def async_update_ha_state(self, force_refresh=False):
                             'https://goo.gl/Nvioub', self.entity_id,
                             end - start)
 
+        def get_overrides(entity_id: str) -> Dict:
+            """Return a dictionary of overrides related to entity_id.
+
+            Whole-domain overrides are of lowest priorities,
+            then regexp on entity ID, and finally exact entity_id
+            matches are of highest priority.
+            """
+            domain_result = regexp_result = exact_result = {}
+            domain = split_entity_id(entity_id)[0]
+            for key, value in _OVERWRITE.items():
+                key_list = [
+                    single_key.strip() for single_key in key.split(',')]
+                if domain in key_list:
+                    domain_result = value
+                if entity_id in key_list:
+                    exact_result = value
+                else:
+                    for pattern in key_list:
+                        if re.fullmatch(pattern, entity_id):
```

Comment:
Also, this feels as if it will be very slow if the user has a lot of customizations. This calculation has to be done constantly in the lifetime of the hass process. What would the impact on general performance be? You're changing a very fast operation (a dictionary lookup) with a couple slow operations (a string split and an indeterminate amount of regular expression matches).